mssqlcli/mssqlutils.py

```diff
@@ -0,0 +1,110 @@
+import itertools
+
+from cli_helpers.tabular_output import TabularOutputFormatter
+from cli_helpers.tabular_output.preprocessors import (align_decimals,
+                                                      format_numbers)
+
+from mssqlcli.encodingutils import utf8tounicode
+from mssqlcli.encodingutils import text_type
+
+
+def has_meta_cmd(query):
+    """Determines if the completion needs a refresh by checking if the sql
+    statement is an alter, create, drop, commit or rollback."""
+    try:
+        first_token = query.split()[0]
+        if first_token.lower() in ('alter', 'create', 'drop'):
+            return True
+    except Exception:
+        return False
+
+    return False
+
+
+def has_change_db_cmd(query):
+    """Determines if the statement is a database switch such as 'use' or '\\c'

```

Comment:
My preference... if a method is only used in one file, define the method that file. In this case, has_changed_db_cmd is only used by MssqlCli, so I think it should be there.

Reply:
This may apply to other methods in this file.

---
In reply to: [171446837](https://github.com/dbcli/mssql-cli/pull/173#discussion_r171446837) [](ancestors = 171446837)